deployability/modules/testing/tests/helpers/actions.py

```diff
@@ -0,0 +1,265 @@
+
+        if self.type_os == 'deb':
+            architecture_suffix['x86_64'] = 'amd64'
+            url = f"https://{self.aws_s3}.wazuh.com/{self.repository}/apt/pool/main/w/wazuh-agent/wazuh-agent_{self.wazuh_version}-{self.wazuh_revision}_{architecture_suffix.get(self.architecture)}.deb"
+            download_command = f'wget {url} -O wazuh-agent_{self.wazuh_version}-{self.wazuh_revision}_{self.architecture}.deb'
+            install_command = f"sudo WAZUH_MANAGER='{self.dependency_ip}' dpkg -i ./wazuh-agent_{self.wazuh_version}-{self.wazuh_revision}_{self.architecture}.deb"
+
+        try:
+            subprocess.run(download_command, shell=True, check=True)
+        except subprocess.CalledProcessError as e:
+            print(f"Error al ejecutar el comando: {e}")
+
+        try:
+            subprocess.run(install_command, shell=True, check=True)
+        except subprocess.CalledProcessError as e:
+            print(f"Error executing the command: {e}")
+
+        post_install_commands = [
+            "sudo systemctl daemon-reload",
+            "sudo systemctl enable wazuh-agent",
+            "sudo systemctl start wazuh-agent"
+        ]
+
+        for command in post_install_commands:
+            try:
+                subprocess.run(command, shell=True, check=True)
+            except subprocess.CalledProcessError as e:
+                print(f"Error executing the command: {e}")
+
+    def _install_windows_agent(self):
+        install_command = f"Invoke-WebRequest -Uri {self.aws_s3}/{self.repository}/windows/wazuh-agent-{self.wazuh_version}-{self.wazuh_revision}.msi -OutFile $env:tmp\\wazuh-agent; msiexec.exe /i $env:tmp\\wazuh-agent /q WAZUH_MANAGER='{self.dependency_ip}' WAZUH_REGISTRATION_SERVER='{self.dependency_ip}'"
+
+        utils.run_command(install_command)
+
+        post_install_command = "NET START WazuhSvc"
+        utils.run_command(post_install_command)
+
+    def _install_macos_agent(self):
+        if self.architecture == 'Intel':
+            command = f"curl -so wazuh-agent.pkg {self.aws_s3}/{self.repository}/macos/wazuh-agent-{self.wazuh_version}-{self.wazuh_revision}.intel64.pkg && echo 'WAZUH_MANAGER='{self.dependency_ip}'' > /tmp/wazuh_envs && sudo installer -pkg ./wazuh-agent.pkg -target /"
+        elif self.architecture == 'Apple':
+            command = f"curl -so wazuh-agent.pkg {self.aws_s3}/{self.repository}/macos/wazuh-agent-{self.wazuh_version}-{self.wazuh_revision}.arm64.pkg && echo 'WAZUH_MANAGER='{self.dependency_ip}'' > /tmp/wazuh_envs && sudo installer -pkg ./wazuh-agent.pkg -target /"
+
+        utils.run_command(command)
+
+        post_install_command = "sudo /Library/Ossec/bin/wazuh-control start"
+        utils.run_command(post_install_command)
+
+class WazuhUninstaller:
+    def __init__(self, os_type, wazuh_version, wazuh_revision, type_os=None):
```

Comment:
Is the revision and version required to uninstall the component? I know that it will be required for the installation, but in this case I think not.

Reply:
version and revision is required in Windows uninstall:

    def _uninstall_windows_agent(self):
        uninstall_command = f"msiexec.exe /x wazuh-agent-{self.wazuh_version}-{self.wazuh_revision}.msi /qn"